metals/src/main/scala/scala/meta/internal/metals/ForwardingMetalsBuildClient.scala

```diff
@@ -139,6 +141,11 @@ final class ForwardingMetalsBuildClient(
             scribe.info(s"time: compiled $name in ${compilation.timer}")
           }
           if (isSuccess) {
+            buildTargetClasses
+              .rebuildIndex(target)
+              .filter(_ => isCurrentlyFocused(target))
```

Comment:
nit: `filter` should generally be avoided on `Future` values because it makes the `Future` fail, it's better to do this instead

```scala
rebuildIndex().foreach { _ =>
  if (foo()) { doBar() }
}
```